barretenberg/cpp/src/barretenberg/stdlib/primitives/field/field.cpp

```diff
@@ -599,17 +643,13 @@ template <typename Builder> void field_t<Builder>::assert_is_zero(std::string co
 
 template <typename Builder> void field_t<Builder>::assert_is_not_zero(std::string const& msg) const
 {
-    if (get_value() == bb::fr(0)) {
-        context->failure(msg);
-        // We don't return; we continue with the function, for debugging purposes.
-    }
 
     if (witness_index == IS_CONSTANT) {
-        ASSERT(additive_constant != bb::fr(0));
+        ASSERT(additive_constant != bb::fr(0), msg);
         return;
     }
 
-    if (get_value() == 0 && context) {
```

Comment:
Should we still have a check on `context`? I know a circuit witness should have a context logically, but just in case?